storage/archive/ha_archive.cc

```diff
@@ -287,7 +287,7 @@ static int archive_discover(handlerton *hton, THD* thd, TABLE_SHARE *share)
   if (!(azopen(&frm_stream, az_file, O_RDONLY|O_BINARY)))
   {
     if (errno == EROFS || errno == EACCES)
-      DBUG_RETURN(my_errno= errno);
+      DBUG_RETURN(HA_ERR_TABLE_READONLY);
```

Comment:
You MUST set my_errno here!

Reply:
Please don't change errno to HA_ERR_CRASHED_ON_USAGE.
This will hide the original error !

Reply:
Instead of changing errno, you can instead change handler::print_error() to give more information about the error